### DISCUSSION TITLE
Remove REcaptcha and propose a real open source solution

### DISCUSSION THREAD

**Discussion by micker** [joomla/joomla-cms#39662] [General]
helloi read a litle about recaptcha google solution solution and its realy not RGPD solution ....
we realy need to change it for open source solution
i search a little and i found this :
- https://github.com/fabianwennink/IconCaptcha-Plugin-jQuery-PHP
- https://github.com/FriendlyCaptcha/friendly-challenge

hrecaptcha isn't realy compilant because we need pro version ... 
i known that only small plugin but regarding joomla philosophy its not good idea to keep a gafam, non RGPD solution in core

**Comment by RussW** [comment]
For the majority of the rest of the world, GDPR is not a concern nor care and Google ReCaptcha is recognisable even to the most basic of internet users, so I see no immediate need to remove it. Having said this, I do recognise the need for alternatives to satisfy the EU and think having such an alternative as another option would be useful.

Personally, I would prefer the "Friendly Captcha" due to its simplicity and ease, especially if it had a less intrusive layout, maybe something like a circular progress meter and middle shield/tick that can be fixed in a bottom corner of a page (similar to Google Invisible ReCaptcha layout)

**Reply by micker** [reply]
GDPR or not can we maintaint a solution that need a consent ? just display it sending user information to google or other ... ?

**Comment by ReLater** [comment]
FriendlyCaptcha seems not to be an alternative because it has not acceptable limits like

![grafik](https://user-images.githubusercontent.com/20780646/220595513-e90247fb-08e4-4ad6-8fd7-cc87f3cc27e5.png)

**Comment by micker** [comment]
friendlycaptcha can be free : https://github.com/FriendlyCaptcha/friendly-lite-server
i found https://mcaptcha.org/ or http://captchas.net/

**Comment by ReLater** [comment]
GDPR compliance is often an hasty argument for this or that solution. In Germany it's forbidden at the moment to connect visitors with external services without consent. We are in an insufficiently well-defined GDPR zone, so to speak.

Therefore I think that none of your suggestions is really GDPR compliant. Or you need a f'ing consent tool. But then I could also use any other captcha on the market or one integrated in Joomla.

> Have a (probably stupid) idea: why not write own solution? Shouldnt be hard.......

It is hard. And I am not stupid enough ;-) I prefere to pay a bit and am happy with the ECC+ plugin of Kubik-Rubik or in exceptional cases with a "real" Joomla captcha plugin like HCaptcha, which is not GDPR compliant.

**Comment by ReLater** [comment]
OK, https://github.com/fabianwennink/IconCaptcha-Plugin-jQuery-PHP/wiki/How-To-Use seems to be a compliant solution ... with a lot of work to integrate it in Joomla with client-side validation.

**Comment by ReLater** [comment]
> Let it be a lot of work.

Go for it, sir! Start the work!

**Reply by brianteeman** [reply]
nothing to do with me. should be relatively simple to integrate and it would be good to see a positive contrinution from you

**Reply by teddyfresco** [reply]
Hi, having such a solution for a long-standing problem would be overkill!!
But please consider the accessibility aspect, as @brianteeman said many times:
https://github.com/fabianwennink/IconCaptcha-Plugin-jQuery-PHP/issues/19
it seems it does not include any feature for visually impaired people, is it even possible to merge it into Joomla! code without accessibility support?
Maybe there's someone who could help Fabian Wennink adding it?

**Reply by teddyfresco** [reply]
I've just found out that Captcheck https://captcheck.netsyms.com/ has got a text mode, I suppose it's ok for accessibility, and it's open source (Is MIT license compatible with GPL?), actively maintained and php/js based. 
Of course it deserves a deeper insight by devs...

**Comment by micker** [comment]
maybe honeypot solution is best first level solution ?
- gdrp
- no cookies
- no need external connexion
https://github.com/paralax/awesome-honeypots
